common/v2/features/SendAssets/components/SignTransactionWallets/Hardware.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect } from 'react';
+
+import { ExtendedAccount as IExtendedAccount, ITxReceipt } from 'v2/types';
+import { makeTransaction } from 'v2/services/EthService';
+import { WalletFactory } from 'v2/services/WalletService';
+import { InlineErrorMsg } from 'v2/components/ErrorMessages';
+
+import { ITxObject, ISignedTx } from '../../types';
+import './Hardware.scss';
+
+export interface IDestructuredDPath {
+  dpath: string;
+  index: number;
+}
+
+export const splitDPath = (fullDPath: string): IDestructuredDPath => {
```

Comment:
const sDPath = (dPath) => {
  const index = dPath.split('/').pop(-1)
  return  { 
    dPath: dPath.substring(0, dPath.length - 1),
    index: parseInt(index, 10)
  };
}

Reply:
And move into `v2/utils`

Reply:
this didn't handle multi-digit indexes `m/44'/60'/0'/544`